sup3r/pipeline/forward_pass.py

```diff
@@ -783,14 +793,24 @@ def __init__(self, strategy, node_index=0):
         self.load_workers = kwargs['load_workers']
         self.output_workers = kwargs['output_workers']
 
-        self.data_handler_class = strategy.data_handler_class
+        fwp_out_shape = len(self.output_features)
+        fwp_out_shape *= np.product(self.data_shape)
+        fwp_out_shape *= self.strategy.s_enhance**2 * self.strategy.t_enhance
+        fwp_out_mem = 4 * fwp_out_shape
+        mem = psutil.virtual_memory()
+        msg = (f'Full size of forward pass output ({fwp_out_mem / 1e9} GB) is '
+               'too large to hold in memory. Run with smaller '
+               'fwp_chunk_shape[2] or smaller spatial extent. ')
+        assert (mem.total - mem.used) > fwp_out_mem, msg
```

Comment:
Recommend this as a warning instead of an assertion error. I'm pretty sure that sometimes python can hold memory and not free it, but it can use this memory for allocating new data. In any case, shouldnt we get an OOM error pretty quickly?

Also recommend you log the mem available in the msg for comparison purposes.

Reply:
Well currently the array to hold the fwp output is declared after the data extraction so it can take some time to get an OOM error. I could just move the declaration though.

Reply:
Actually I probably don't want to move the declaration bc that will restrict the mem available for the data extraction.

Reply:
Okay, well i dont love it but lets see how it goes haha

Reply:
This is really just to check if the data array that holds the fwp output is bigger than the total mem. If we are declaring an 900 GB array on a 800 GB machine itll fail no matter what magic python does

Reply:
*Bigger than the _available_ memory. psutil is not perfect and I'd always rather be able to try stuff with a warning instead of the code not letting me.

Reply:
Yeah I could change it to total mem instead of available?

Reply:
I like that!